src/cell.cpp

```diff
@@ -334,10 +334,10 @@ double Cell::temperature(int32_t instance) const
 void Cell::set_temperature(double T, int32_t instance, bool set_contained)
 {
   if (settings::temperature_method == TemperatureMethod::INTERPOLATION) {
-    if (T < data::temperature_min) {
+    if (T < data::temperature_min - settings::temperature_tolerance) {
       throw std::runtime_error {"Temperature is below minimum temperature at "
                                 "which data is available."};
-    } else if (T > data::temperature_max) {
+    } else if (T > data::temperature_max + settings::temperature_tolerance) {
       throw std::runtime_error {"Temperature is above maximum temperature at "
                                 "which data is available."};
```

Comment:
It could be useful to let the user know the closest temperature at which data is available and/or the closest temperature that would not trigger an error. They could probably look it up, but this error message could help get there faster.

Reply:
Hi Lewis! Thanks for the comment. I just pushed printing of the nearest temperature of the bounds within the error message.